solr/app.py

```diff
@@ -17,10 +19,37 @@ def create_app():
     app.url_map.strict_slashes = False
     Consul(app)  # load_config expects consul to be registered
     load_config(app)
+    if config:
+        app.config.update(config)
+        
+    db.init_app(app)
     logging.config.dictConfig(
         app.config['SOLR_SERVICE_LOGGING']
     )
     
+    ## pysqlite driver breaks transactions, we have to apply some hacks as per
```

Comment:
I'm not sure I understand then.  If your point was to have a dedicated
class (like the linked example) then that would just move the code
elsewhere (and not help make the code simpler). If it is about removing
support for sqlite, because we have to have this hack to support
transactions, that would seem too strict to me. Sure,  in production sqlite
sucks, but we never set postgres as the only store. More over, I would be
careful to use product specific dialect like in the example. If postgres
was the only engine, ones main reason for using sqlalchemy is gone.
On Jun 13, 2016 10:15 PM, "Jonny Elliott" <notifications@github.com> wrote:

> In solr/app.py
> <https://github.com/adsabs/solr-service/pull/14#discussion_r66869568>:
>
> >      logging.config.dictConfig(
> >          app.config['SOLR_SERVICE_LOGGING']
> >      )
> >
> > +    ## pysqlite driver breaks transactions, we have to apply some hacks as per
>
> Yeah, I understood what it's doing, I just didn't think it necessarily
> needs to be here. More likely than not we will go with postgres, so it just
> adds a bit of overhead if someone is reading this. But I have no strong
> preference for it staying or going.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/adsabs/solr-service/pull/14/files/263c938fe9d884068f53ea61a9c8add5faf002f1#r66869568>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAZIkhxmg6_grmBG3FpwkksB5-vQ8HXPks5qLchbgaJpZM4I0SZd>
> .
>